tools/android/packaging/xbmc/src/org/xbmc/kodi/XBMCJsonRPC.java.in

```diff
@@ -0,0 +1,647 @@
+package org.xbmc.kodi;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.protocol.HTTP;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.SearchManager;
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.database.MatrixCursor;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.provider.BaseColumns;
+import android.util.Log;
+
+public class XBMCJsonRPC
+{
+  public final static String APP_NAME = "@APP_NAME@ Search";
+  public final static String COLUMN_FULL_PATH = "COLUMN_FULL_PATH";
+  public final static String COLUMN_BASE_PATH = "COLUMN_BASE_PATH";
+  public final static String COLUMN_FILENAME = "COLUMN_FILENAME";
+  public final static String COLUMN_TITLE = "COLUMN_TITLE";
+  public final static String COLUMN_TAGLINE = "COLUMN_TAGLINE";
+  public final static String COLUMN_THUMB = "COLUMN_THUMB";
+  public final static String COLUMN_FANART = "COLUMN_FANART";
+  public final static String COLUMN_ID = "COLUMN_ID";
+  public final static String COLUMN_VIEW_PROGRESS = "COLUMN_VIEW_PROGRESS";
+  public final static String COLUMN_RECOMMENDATION_REASON = "COLUMN_RECOMMENDATION_REASON";
+
+  public final static String REQ_ID_MOVIES = "1";
+  public final static String REQ_ID_SHOWS = "2";
+  public final static String REQ_ID_ALBUMS = "3";
+  public final static String REQ_ID_ARTISTS = "4";
+
+  private static String TAG = "@APP_NAME@json";
+
+  private String m_jsonURL = "http://localhost:8080";
```

Comment:
Separate app, because a) you can improve it independet of release cycles (which might be needed), b) can make it autolaunch Kodi on startup, c) find a pure Java dev as maintainer, d) more easily share code with Kore, ...

At least in my opinion this could work very well as standalone companion app. If you guys think it's better suited as part of Kodi itself, I'm not against doing it that way, but given that we remote access the main Kodi process anyways and that it might be a bit tricky to add a Android based config UI for ports etc (which appears to be needed), it just feels wrong as part of the main app. If we won't need any user config options, then I'd of course prefer it being part of Kodi itself.